cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -229,6 +234,10 @@ public SparkClassRewriter(Function<String, InputStream> resourceLookup, boolean
       // Rewrite the Spark OutputMetrics to skip overwriting bytes written metrics with 0.
       return rewriteOutputMetrics(SPARK_OUTPUT_METRICS, input);
     }
+    if (className.startsWith(CHECKPOINT_WRITE_HANDLER_TYPE.getClassName()) && rewriteCheckpointTempFileName) {
```

Comment:
I would check the boolean first for a faster short circuit

Reply:
Good point. I wanted to keep it consistent with https://github.com/cdapio/cdap/blob/develop/cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java#L188

Reply:
Maybe worth changing both places

Reply:
Done!